transports/uds/src/lib.rs

```diff
@@ -103,7 +114,7 @@ impl Transport for UdsConfig {
     fn dial(self, addr: Multiaddr) -> Result<Self::Dial, TransportError<Self::Error>> {
         if let Ok(path) = multiaddr_to_path(&addr) {
             debug!("Dialing {}", addr);
-            Ok(UnixStream::connect(&path))
+            Ok(Box::pin(async move { UnixStream::connect(&path).await }))
```

Comment:
The `Future` returned by `connect` unfortunately borrows `path`. This `async` block is here in order to bypass this problem.